server/internal/request.go

```diff
@@ -100,6 +100,7 @@ func getAdapterRequest[Config any](
 		return nil, nil, adapterErr
 	}
 
+	adapterRequest.DatasourceID = req.Datasource.Id
```

Comment:
This is already checked for nil further up:
https://github.com/SGNL-ai/adapter-framework/pull/42/commits/cda92e309182dbec842d1b0826cbaf1a5928978d#diff-231caa878672fe80dbefaa1069dd31fd21f4a674d619fa66c54c54f95b8faa55R64-R67